src/test/regress/sql/multi_extension.sql

```diff
@@ -78,6 +78,71 @@ ALTER EXTENSION citus UPDATE TO '6.1-15';
 ALTER EXTENSION citus UPDATE TO '6.1-16';
 ALTER EXTENSION citus UPDATE TO '6.1-17';
 ALTER EXTENSION citus UPDATE TO '6.2-1';
+
+-- Test non-distributed queries work even in version mismatch
+SET citus.enable_version_checks TO 'true';
+
+-- Test CREATE TABLE
+CREATE TABLE version_mismatch_table(column1 int);
+
+-- Test COPY
+\copy version_mismatch_table FROM STDIN;
+0
+1
+2
+3
+4
+\.
+
+-- Test INSERT
+INSERT INTO version_mismatch_table(column1) VALUES(5);
+ 
+-- Test SELECT
+SELECT * FROM version_mismatch_table ORDER BY column1;
+
+-- Test SELECT from pg_catalog
+SELECT d.datname as "Name",
+       pg_catalog.pg_get_userbyid(d.datdba) as "Owner",
+       pg_catalog.array_to_string(d.datacl, E'\n') AS "Access privileges"
+FROM pg_catalog.pg_database d
+ORDER BY 1;
+
+-- We should not distribute table in version mistmatch
+SELECT create_distributed_table('version_mismatch_table', 'column1');
+
+-- This function will cause fail in next ALTER EXTENSION
+CREATE OR REPLACE FUNCTION pg_catalog.citus_table_size(table_name regclass)
+RETURNS bigint LANGUAGE plpgsql
+AS $function$
+DECLARE
+        table_size bigint;
+BEGIN
+        PERFORM master_update_shard_statistics(shardid)
+        FROM pg_dist_shard
+        WHERE logicalrelid = table_name;
+
+        SELECT sum(shard_size) INTO table_size
+        FROM (
+                SELECT max(shardlength) AS shard_size
+                FROM pg_dist_shard JOIN pg_dist_shard_placement USING (shardid)
+                WHERE logicalrelid = table_name AND shardstate = 1
+                GROUP BY shardid
+        ) shard_sizes;
+
+        RETURN table_size;
+END;
```

Comment:
It causes next ALTER EXTENSION citus UPDATE TO '6.2-2'; fail. Because we also create a function with same name. It tests we can continue to operate when altering extension failed. I added it because of [this issue](https://github.com/citusdata/citus/issues/1422).